Source/Scene/Model.js

```diff
@@ -2058,8 +2069,13 @@ define([
                 pickFS = modifyShaderForClippingPlanes(pickFS, clippingPlaneCollection);
             }
 
-            pickVS = ModelUtility.modifyVertexShaderForLogDepth(pickVS, toClipCoordinatesGLSL);
-            pickFS = ModelUtility.modifyFragmentShaderForLogDepth(pickFS);
+            // Internet Explorer seems to have problems with discard after too many levels of indirection:
+            // https://github.com/AnalyticalGraphicsInc/cesium/issues/6575.
+            // In this case, log depth modifications are empty wrappers and can be omitted.
+            if (!FeatureDetection.isInternetExplorer()) {
```

Comment:
Why not detect if log depth is supported instead of checking IE specifically?

Reply:
It's an IE bug more than a log depth bug, so I think this is more descriptive.

We could also check inside the modified functions for log depth support and just not modify the shader, but I'm not sure if there was some reason that wasn't happening already.
Maybe the current design was to keep log depth toggling to a single shader define for simplicity?
We'd also have to wire access to the `context` down to the modifier functions too.

Reply:
Did you try other browsers/OS's without log depth just to be sure?  My main concern is that we don't know WHY it's crashing in IE, right?  We just assume it's a bug in IE and not something non-standard in our own code.

Reply:
Clipping planes and picking with clipping planes seem to work fine on Chrome on my Android phone, which doesn't have the extensions needed for log depth.

Reply:
OK, cool thanks for checking.